spikes/ktor-test-server/src/main/kotlin/dextest/plugins/Auth.kt

```diff
@@ -0,0 +1,128 @@
+package dextest.plugins
+
+import io.ktor.server.application.*
+import io.ktor.server.auth.*
+import kotlinx.serialization.Serializable
+
+import com.google.auth.oauth2.TokenVerifier
+
+import com.nimbusds.jwt.SignedJWT
+import com.nimbusds.jose.crypto.RSASSAVerifier
+import com.nimbusds.jose.jwk.JWK
+import com.nimbusds.jose.jwk.JWKSet
+import com.nimbusds.jose.jwk.RSAKey
+import org.json.JSONObject
+import java.net.URL
+import java.security.interfaces.RSAPublicKey
+
+
+@Serializable
+data class Claims(val scope: String)
+
+@Serializable
+data class IntrospectionResponse(val active: Boolean, val scope: String)
+
+data class AuthConfig(
+    val issuerUrl: String, val introspectionUrl: String, val requiredScopes: String?
+)
+
+// Dummy appConfig, replace with actual config loading logic
+val authConfig = AuthConfig(
+    issuerUrl = "https://apigw-stg.cdc.gov:8443",
+    introspectionUrl = "https://your-introspection-url",
+    requiredScopes = "dex:upload"
+)
+
+fun Application.configureAuth() {
```

Comment:
Please test and make sure this plays nicely with the graphQL module. The graphQL module has its own way of doing auth so my only concern at this point is that these two will integrate successfully.

Reply:
@mkrystof are you referring the current `Auth0` package setup in the `Graphql.kt` file? The part that looks like this: 

`    if (securityEnabled) {
        install(Authentication) {
            jwt {
                jwt("auth-jwt") {
                    realm = myRealm
                    verifier(
                        JWT
                            .require(Algorithm.HMAC256(Base64.getDecoder().decode(secret)))
                            .withAudience(audience)
                            .withIssuer(issuer)
                            .build()
                    )
                    validate { credential ->
                        if (credential.payload.getClaim("username").asString() != "") {
                            JWTPrincipal(credential.payload)
                        } else {
                            logger.error("username missing from JWT claims, denying the token")
                            null
                        }
                    }
`